assignment_2.java

```diff
@@ -0,0 +1,144 @@
+                String replacement = scanner.nextLine();
+                System.out.println("Result: " + replace(input, target, replacement));
+                break;                     
+            case 7:
+                System.out.println("Max repeating character: " + maxRepeatingCharacter(input));
+                break;
+            case 8:
+                System.out.println("Sorted string: " + sort(input));
+                break;
+            case 9:
+                System.out.println("Enter shift amount:");
+                int shiftAmount = scanner.nextInt();
+                scanner.nextLine();  // Consume newline character
+                System.out.println("Result: " + shift(input, shiftAmount));
+                break;
+            case 10:
+                System.out.println("Reversed string: " + reverse(input));
+                break;
+            default:
+                System.out.println("Invalid choice.");
+        }
+        System.out.print(constant.TRY_AGAIN);
+        tryagain = scanner.nextLine();
+    }
+
+    System.out.println("Program terminated.");
+    scanner.close(); } 
+
+    // Append a string
+    public static String Append_String(String input, String appendStr) {
+    	input+= appendStr;
+        System.out.println("Given String: " + input);
+        return input; // Return the updated string
+    }
+
+    // Replace a substring with another substring
+    public static String replace(String str, String target, String replacement) {
+        return str.replace(target, replacement);
+    }
+
+    
+    // Find the maximum repeating character
+    public static char maxRepeatingCharacter(String str) {
+        int[] charCount = new int[256]; 
+        for (int i = 0; i < str.length(); i++) {
+            charCount[str.charAt(i)]++;
+        }
+
+        int maxCount = -1;
+        char result = '\0';
+        for (int i = 0; i < charCount.length; i++) {
+            if (charCount[i] > maxCount) {
+                maxCount = charCount[i];
+                result = (char) i;
+            }
+        }
+        return result;
+    }
+
+    // Sort the characters of the string
+    public static String sort(String str) {
+        char[] chars = str.toCharArray();
+        for (int i = 0; i < chars.length - 1; i++) {
+            for (int j = i + 1; j < chars.length; j++) {
+                if (chars[i] > chars[j]) {
+                    char temp = chars[i];
+                    chars[i] = chars[j];
+                    chars[j] = temp;
+                }
+            }
+        }
+        return new String(chars);
+    }
+
+    // Shift characters in the string by a given amount
+    public static String shift(String str, int shiftAmount) {
+        int length = str.length();
+        shiftAmount = (shiftAmount % length + length) % length; // Handle negative shift amounts
+        return str.substring(length - shiftAmount) + str.substring(0, length - shiftAmount);

```

Comment:
Ensure shiftAmount is not greater than length to avoid StringIndexOutOfBoundsException.